Wikipedia/Code/WMFArticleImageGalleryDataSource.m

```diff
@@ -26,7 +27,28 @@ @implementation WMFArticleImageGalleryDataSource
 @dynamic emptyView;
 
 - (instancetype)initWithArticle:(MWKArticle*)article {
-    self = [super wmf_initWithItemsAndReverseIfNeeded:article.images.uniqueLargestVariants];
+    
+    NSArray* images = [article.images.uniqueLargestVariants bk_select:^BOOL (MWKImage* image) {
+        // Keep image if it's the article image even if we can't determine its size - we
+        // always want to show article image as first "lead" image by gallery.
+        if ([article.image isEqualToImage:image]) {
+            return YES;
+        }
+        
+        if (!image.width || !image.height) {
+            // HAX: if this image MWKImage record doesn't have a width and height value (because
+            // it wasn't determined by parsing it from the article HTML's image url) see if the
+            // cache can tell us the size.
+            UIImage* img = [[WMFImageController wmf_appImageCache] imageFromDiskCacheForKey:image.sourceURLString];
```

Comment:
FYI, this could be very expensive (reading the entire image from disk to get the size). you're doing this synchronously, on the main thread. are there any known cases where images w/o given widths/heights are actually above the threshold _and_ worth including in the gallery?